internal/pkg/service/stream/migrate/source/source.go

```diff
@@ -0,0 +1,190 @@
+	TableID     string    `json:"tableId"`
+	Incremental bool      `json:"incremental"`
+	Columns     []Columns `json:"columns"`
+}
+type Conditions struct {
+	Count int    `json:"count"`
+	Size  string `json:"size"`
+	Time  string `json:"time"`
+}
+type Exports struct {
+	ID         string     `json:"id"`
+	ReceiverID string     `json:"receiverId"`
+	Name       string     `json:"name"`
+	Mapping    Mapping    `json:"mapping"`
+	Conditions Conditions `json:"conditions"`
+}
+type Receiver struct {
+	ID          string    `json:"id"`
+	URL         string    `json:"url"`
+	Name        string    `json:"name"`
+	Description string    `json:"description"`
+	Exports     []Exports `json:"exports"`
+}
+
+type RequestConfig struct {
+	method string
+	host   string
+	token  string
+	path   string
+	body   io.Reader
+	logger log.Logger
+}
+
+func New(method string, host string, token string, path string, body io.Reader, l log.Logger) RequestConfig {
+	return RequestConfig{
+		method: method,
+		host:   host,
+		token:  token,
+		body:   body,
+		path:   path,
+		logger: l,
+	}
+}
+
+func FetchBufferReceivers(ctx context.Context, host string, token string, logger log.Logger) (*Receivers, error) {
+	return fetchDataFromBuffer(ctx, New(
+		"GET",
+		"https://"+host,
+		token,
+		receiversBufferPath,
+		nil,
+		logger,
+	))
+}
+
+func fetchDataFromBuffer(ctx context.Context, reqConfig RequestConfig) (*Receivers, error) {
+	resp, err := newHTTPRequest(ctx, reqConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.Errorf(`unexpected status code %d from %s`, resp.StatusCode, reqConfig.host)
+	}
+
+	var result *Receivers
+
+	err = json.NewDecoder(resp.Body).Decode(&result)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func (r *Receiver) CreateSource(ctx context.Context, token string, host string, wg *sync.WaitGroup, l log.Logger) error {
+	defer wg.Done()
+
+	// Set a payload to create source
+	body, err := r.createSourcePayload()
+	if err != nil {
+		return err
+	}
+
+	streamHost := strings.Replace(host, bufferPrefix, streamPrefix, 1)
+
+	// Request to create source
+	resp, err := newHTTPRequest(ctx, RequestConfig{
+		token:  token,
+		method: "POST",
+		path:   createSourcePath,
+		host:   "https://" + streamHost,
+		body:   body,
+		logger: l,
+	})
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusAccepted:
+		l.Infof(ctx, `Source %s created`, r.Name)
+	case http.StatusConflict:
+		l.Infof(ctx, `Source "%s" already exists in the branch\n`, r.Name)
+	case http.StatusUnprocessableEntity:
+		l.Infof(ctx, `Maximum number of sources per project is 100.`)
+	default:
+		l.Infof(ctx, "Unexpected status code: %v for source %s.", resp.StatusCode, r.ID)
+	}
+	return nil
+}
+
+func newHTTPRequest(ctx context.Context, c RequestConfig) (*http.Response, error) {
+	url := fmt.Sprintf("%s%s", c.host, c.path)
```

Comment:
We should enable linter with reworking these `fmt.Sprintf` statements to construct URL. 
More convinient way to do it is to actually use `url` package, Then you use the constructor `url.URL: {Host: c.host, Path: c.path}` Instead of `fmt.Sptrintf`. :sweat_smile: